runtime/memprof.c

```diff
@@ -2275,9 +2276,18 @@ CAMLprim value caml_memprof_stop(value unit)
 {
   memprof_domain_t domain = Caml_state->memprof;
   CAMLassert(domain);
-  CAMLassert(domain->current);
-  value config = thread_config(domain->current);
+  memprof_thread_t thread = domain->current;
+  CAMLassert(thread);
+
+  /* Final attempt to run allocation callbacks. */
+  if (!thread->suspended) {
+    update_suspended(domain, true);
+    value res = entries_run_callbacks_exn(thread, &thread->entries);
+    update_suspended(domain, false);
+    (void) caml_raise_if_exception(res);
+  }
```

Comment:
You seems careful not to call `caml_memprof_run_callbacks_exn` (another choice could be `caml_process_pending_actions`). What is the reason for this choice?

Reply:
This was just a judgement call. I mainly want to run any pending allocation callbacks here, to avoid them being discarded. Pending allocation callbacks are all on the `thread->entries` table. If I call`caml_memprof_run_callbacks_exn`, then other entries tables for this domain will be processed too, including (at present) the `domain->entries` table before the `thread->entries` callbacks. This may prevent the `thread->entries` callbacks from running, for example if one of the `domain->entries` callbacks raises an exception.
This is really just an aesthetic choice, as of course those other callbacks might run at any time.

Reply:
Ok, it's your call.

Reply:
I've added a comment.